gammapy/makers/safe.py

```diff
@@ -274,20 +270,15 @@ def make_mask_bkg_clip(self, dataset):
             Safe data range mask.
         """
 
-        bkg = dataset.background.data
+        bkg = dataset.background.data.copy()
+        bkg[bkg == 0.0] = np.nan
         mask = np.isfinite(bkg)
         if self.n_95percentile is not None:
-            for k in range(bkg.shape[0]):
-                data = bkg[k, :, :]
-                thr = self.n_95percentile * np.nanpercentile(data[data > 0], 95)
-                mask[k, :, :] = np.abs(bkg[k, :, :]) < thr
-        if self.force_clipping:
-            dataset.background.data[~mask] = 0.0
+            thr = self.n_95percentile * np.nanpercentile(bkg, 95, axis=(1,2))
+            mask = np.abs(bkg) < thr[:, None, None]
```

Comment:
One minor comment here: you can use `np.nanpercentile(..., keepdims=True)` to avoid the `[:, None, None]`